### DISCUSSION TITLE
Allow setting different Policy directory for each Panel provider

### DISCUSSION THREAD

**Discussion by Sh1d0w** [filamentphp/filament#8725] [Ideas]
Our use case is that we have multiple Panel providers, which also have different authentication providers (meaning different tables from the database).

Everything works correctly except that both panels use the same Policies for checking the resource authorisation, thus throwing an exception because if the types passed in the policy methods.

It would be nice to expose a method on the panel providers that allows us to configure the namespace/directory of the policies that have to be loaded for that panel.

**Comment by leandrocfe** [comment]
A different way would be to access the current panel `Filament::getCurrentPanel()->getId()` to apply different permissions.

**Comment by Sh1d0w** [comment]
Actually I've found a workaround, not sure if it have to be documented but in each panel provider you can define a gate policy resolver in the boot method like this:

```
/**
     * Register any authentication / authorization services.
     */
    public function boot(): void
    {
        Gate::guessPolicyNamesUsing(function (string $modelClass) {
            // your logic here
        });
    }
 ```

This allows for customising the resolver of the policies per panel, which is exactly what I've needed.
